LinearRegression/LinearRegressionWPF/Views/ImageToVectorTab.xaml.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Drawing;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using Brushes = System.Windows.Media.Brushes;
+using LinearRegressionBackend.DataProvider;
+using LinearRegressionWPF.ViewModels;
+
+namespace LinearRegressionWPF.Views
+{
+    /// <summary>
+    /// Interaction logic for ImageToVectorTab.xaml
+    /// </summary>
+    public partial class ImageToVectorTab
+    {
+
+        //initialize the dataconverter object 
+        DataConverter dataConverter = new DataConverter();
+        ImageToVectorViewModel imageToVectorViewModel = new ImageToVectorViewModel(); 
+        public ImageToVectorTab()
+        {
+            InitializeComponent();
+
+            DataContext = new ImageToVectorViewModel();
+            // set drawing color to black 
+            MyCanvas.DefaultDrawingAttributes.Color = Colors.Black;
+            // activate the ue of customized cursors 
+            MyCanvas.UseCustomCursor = true;
+            // border indicating the active tool of drawing 
+            brushbrd.BorderBrush = Brushes.Blue;
+            //initialize the cursor of the inkcanvas to pen cursor
+            MyCanvas.Cursor = Cursors.Pen;
+
+           
+        }
+
+
+
+      
+
+    
+        private void BrushBtnClick(object sender, MouseButtonEventArgs e)
+        {
+            /* logic for switching between the button and the eraser */
+            brushbrd.BorderBrush = Brushes.Blue;
+            MyCanvas.EditingMode = InkCanvasEditingMode.Ink;
+            MyCanvas.Cursor = Cursors.Pen;
+            erasebrd.BorderBrush = Brushes.Gray;
+        }
+
+        private void EreaseBtnClick(object sender, MouseButtonEventArgs e)
+        {
+            /* logic for switching between the button and the eraser */
+            erasebrd.BorderBrush = Brushes.Blue;
+            MyCanvas.EditingMode = InkCanvasEditingMode.EraseByPoint;
+            MyCanvas.Cursor = System.Windows.Input.Cursors.Cross;
+            brushbrd.BorderBrush = Brushes.Gray;
+        }
+
+        private void ClearBtnClick(object sender, RoutedEventArgs e)
+        {
+            MyCanvas.Strokes.Clear();
+        }
+
+    
+        private void FourByFourBtnClick(object sender, RoutedEventArgs e)
+        {
+            GetMatrix(4, 4);
+        }
+  
+
+        /// <summary>
+        /// print the matrix on the interface an ivestigate visualy 
+        /// </summary>
+        /// <param name="src"> get Image source to show the matrix in the interface </param>
+        public void DrawMatrix(ImageSource src)
```

Comment:
Instead of drawing to the UI, save to a file.
Example of a file saving 4x4 image:
0.75 0.41 0.75 0.31
0.69 0.66 0.99 0.12
0.05 0.47 0.27 0.31
0.07 0.53 0.77 0.58